text/0000-remove-stacks-mixins.md

```diff
@@ -0,0 +1,110 @@
+
+# Motivation
+[motivation]: #motivation
+
+The "stack" and "mixin" concepts add unnecessary complexity to the project and make it difficult for new users and contributors to understand how buildpacks work. Compatibility guarantees that are strongly enforced by the stack contract could be replaced with metadata validations and warnings.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Summary of changes:
+- Replace mixins with a [CycloneDX](https://cyclonedx.org)-formatted list of packages.
+- Replace stack metadata (including stack IDs) with canonical OS metadata.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Base Image Metadata
+
+Instead of a stack ID, runtime and build-time base images are labeled with the following canonicalized metadata:
+- OS (e.g., "linux", `$GOOS`)
+- Architecture (e.g., "x86_64", `$GOARCH`)
+- Distribution (optional) (e.g., "ubuntu", `$ID`)
+- Version (optional) (e.g., "18.04", `$VERSION_ID`)
+
+For Linux-based images, each field should be canonicalized against values specified in `/etc/os-release` (`$ID` and `$VERSION_ID`).
+
+The `stacks` list in `buildpack.toml` is replaced by a `platforms` list, where each entry corresponds to a different buildpack image that is exported into a [manifest index](https://github.com/opencontainers/image-spec/blob/master/image-index.md). Each entry may contain multiple valid values for Distribution and/or Version, but only a single OS and Architecture. Each entry may also contain a list of package names (as PURL URLs without versions or qualifiers) that specify detect-time and build-time (but not runtime) OS package dependencies. Packages may be specified for all Versions of a distribution, or for a specific Version. Buildpacks may express **runtime** OS package dependencies during detection (see "Runtime Base Image Requirements" below).
+
+App image builds fail if the build image and selected run image have mismatched metadata. We may introduce flags or additional labels to skip this validation (e.g., for cross-compilation or minimal runtime base images). An image without a specified Distribution is compatible with images specifying any Distribution. An image specifying a Distribution without a Version is compatible with images specifying any Versions of that Distribution.
+
+When an app image is rebased, `pack rebase` will fail if the new run image and previous run image have mismatched metadata. This check may be skipped for Distribution and Version by passing a new `--force` flag to `pack rebase`.
+
+#### Example: buildpack.toml `platforms` table
+
+```toml
+[[platforms]]
+os = "linux"
+arch = "x86_64"
+[[platforms.distros]]
+name = "ubuntu"
+[[platforms.distros.versions]]
+version = "18.04"
+packages = ["pkg:deb/ubuntu/curl"]
+[[platforms.distros.versions]]
+version = "20.04"
+packages = ["pkg:deb/ubuntu/curl2"]
+
+[[platforms]]
+os = "linux"
+arch = "x86_64"
```

Comment:
```suggestion
os = "linux"
arch = "arm64"
```
I think this needs to have a different os/arch combo than the other entries right? Otherwise selecting the correct buildpackage from the manifest list will require extra logic.

Reply:
The intent was to allow different buildpacks to be built for different distribution with the same os/arch. Manifest Indexes allow this. We need custom logic to use buildpackages anyways.

Reply:
Added extra arm example